Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -537,12 +578,38 @@ protected override void ProcessRecord()
         {
             base.ProcessRecord();
 
+            // If we are in Google Cloud Storage Provider location, resolve the path to get possible bucket name and prefix.
+            string[] bucketAndPrefix = GetBucketAndPrefix();
+            string gcsProviderPrefix = null;
+            if (bucketAndPrefix != null && bucketAndPrefix.Length > 0)
+            {
+                Bucket = Bucket ?? bucketAndPrefix[0];
+                if (bucketAndPrefix.Length == 2)
+                {
+                    gcsProviderPrefix = bucketAndPrefix[1];
```

Comment:
You use the same general logic above to get the prefix and bucket. Maybe make a shared function?